### DISCUSSION TITLE
Symbol Lookup Error

### DISCUSSION THREAD

**Discussion by ritamcj** [idaholab/moose#18167] [Q&A General] [answered]
Hi everyone,
I am working with the CoupledGradient API in Moose and attempting to calculate grad(Fp) in the tensor mechanics module where Fp is the plastic deformation gradient. I duplicated the CoupledGradient function and all associated functions in Coupleable.C to try read a double variable instead of a string. I left all original functions unchanged and everything compiled nicely. But when I run my input file, I see a symbol lookup error in the heat conduction module as attached. I checked the error with C++filt and it showed "non-virtual thunk to SideIntegralPostprocessor::getValue()". I am unable to understand this error. I would appreciate any guidance to solve this problem.
Thanks and regards,
Ritam Chatterjee

![Screenshot from 2021-06-25 09-44-19](https://user-images.githubusercontent.com/81842276/123374574-f06d9b80-d5a4-11eb-9b65-db513be7143e.png)
![Screenshot from 2021-06-25 10-24-14](https://user-images.githubusercontent.com/81842276/123374984-991bfb00-d5a5-11eb-9a1f-cd72b5a0bbff.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

first thing I would try would be to clean and rebuild just in case something did not get updated from before the modifications. So 
‘make clobberall’ then ‘make’

then if that doesn’t work, did you make sure to have an implementation for every single routine you added in coupleable.h ? You did not add any pure virtual ones?

best 
Guillaume

**Reply by ritamcj** [reply]
Hi Guillaume,
Thank you for your response. I tried the clobber make method in both tensor mechanics and heat conduction modules but the error persists. I did not add any pure virtual functions in Coupleable.h but I commented out some statements in my new functions that I felt were not critical to the new approach. Only the old, unaltered functions are being called everywhere in different modules. Can i do something with getValue() in SideIntegralPostProcessor.C that would resolve this?
Ritam

**Reply by ritamcj** [reply]
Hi Guillaume,
I tried clobber make in the combined module and it worked! Many thanks for your suggestion.
Best regards,
Ritam
